macOS/DuckDuckGo/CrashReports/Model/CrashReportReader.swift

```diff
@@ -17,68 +17,97 @@
 //
 
 import Foundation
+import Common
 
 final class CrashReportReader {
 
-    static let displayName = Bundle.main.displayName
-    static let vpnExtensionDisplayName = "com.duckduckgo.macos.vpn.network-extension"
+    static func validBundleIdentifiers() -> [String] {
+        return [
+            Bundle.main.bundleIdentifier,
+            Bundle.main.vpnMenuAgentBundleId,
+            Bundle.main.vpnSystemExtensionBundleId,
+            Bundle.main.vpnProxyExtensionBundleId,
+            Bundle.main.dbpBackgroundAgentBundleId
+        ].compactMap(\.self)
+    }
+
+    private let fileManager: FileManager
+    private let validBundleIdentifierProvider: () -> [String]
+    private let dateProvider: () -> Date
+
+    init(fileManager: FileManager = .default,
+         validBundleIdentifierProvider: @escaping () -> [String] = CrashReportReader.validBundleIdentifiers,
+         dateProvider: @escaping () -> Date = Date.init) {
+        self.fileManager = fileManager
+        self.validBundleIdentifierProvider = validBundleIdentifierProvider
+        self.dateProvider = dateProvider
+    }
 
     func getCrashReports(since lastCheckDate: Date) -> [CrashReport] {
         var allPaths: [URL]
 
         do {
-            allPaths = try FileManager.default.contentsOfDirectory(at: FileManager.userDiagnosticReports, includingPropertiesForKeys: nil)
+            allPaths = try fileManager.contentsOfDirectory(at: FileManager.userDiagnosticReports, includingPropertiesForKeys: nil)
         } catch {
             assertionFailure("CrashReportReader: Can't read content of diagnostic reports \(error.localizedDescription)")
             return []
         }
 
         do {
-            let systemPaths = try FileManager.default.contentsOfDirectory(at: FileManager.systemDiagnosticReports, includingPropertiesForKeys: nil)
+            let systemPaths = try fileManager.contentsOfDirectory(at: FileManager.systemDiagnosticReports, includingPropertiesForKeys: nil)
             allPaths.append(contentsOf: systemPaths)
         } catch {
             assertionFailure("Failed to read system crash reports: \(error)")
         }
 
         let filteredPaths = allPaths.filter({
-            isCrashReportPath($0) && belongsToThisApp($0) && isFile(at: $0, newerThan: lastCheckDate)
```

Comment:
`belongsToThisApp` isn't needed any more since `isCrashReportPath` handles filtering of reports that don't look like they come from our app.